src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs

```diff
@@ -23,30 +29,172 @@ public partial class SplashScreen : Window
 
         private long totalLoadingTime;
 
-        private DirectoryInfo webBrowserUserDataFolder;
+        private readonly DirectoryInfo webBrowserUserDataFolder;
 
         internal Action<bool> RequestLaunchDynamo;
         internal Action<string> RequestImportSettings;
         internal Func<bool> RequestSignIn; 
         internal Func<bool> RequestSignOut;
         internal WebView2 webView;
+        private DynamoView dynamoView;
+        private AuthenticationManager authManager;
+        private readonly string ASMPath;
+        private readonly string CERPath;
+        private readonly string commandFilePath;
+        private readonly HostAnalyticsInfo hostAnalyticsInfo;
+
+        /// <summary>
+        /// Dynamo View Model reference
+        /// </summary>
+        public DynamoViewModel viewModel;
 
         /// <summary>
         /// Constructor
         /// </summary>
-        public SplashScreen(string userDataDirectory)
+        public SplashScreen(HostAnalyticsInfo info = new HostAnalyticsInfo(), string asmPath ="", string cerPath ="", string cmdFilePath ="")
         {
+            ASMPath = asmPath;
+            CERPath = cerPath;
+            commandFilePath = cmdFilePath;
+            hostAnalyticsInfo = info;
             InitializeComponent();
 
             loadingTimer = new Stopwatch();
             loadingTimer.Start();
 
-            //When executing Dynamo as Sandbox or inside any host like Revit, FormIt, Civil3D the WebView2 cache folder will be located in the AppData folder
-            var userDataDir = new DirectoryInfo(userDataDirectory);
+            // When executing Dynamo as Sandbox or inside any host like Revit, FormIt, Civil3D the WebView2 cache folder will be located in the AppData folder
+            var userDataDir = new DirectoryInfo(GetUserDirectory());
             webBrowserUserDataFolder = userDataDir.Exists ? userDataDir : null;
 
             webView = new WebView2();
             AddChild(webView);
+            webView.NavigationCompleted += WebView_NavigationCompleted;
+            DynamoModel.RequestUpdateLoadBarStatus += DynamoModel_RequestUpdateLoadBarStatus;
+            // Bind event handlers
+            RequestLaunchDynamo = LaunchDynamo;
+            RequestImportSettings = ImportSettings;
+            RequestSignIn = SignIn;
+            RequestSignOut = SignOut;
+        }
+
+        private void WebView_NavigationCompleted(object sender, CoreWebView2NavigationCompletedEventArgs e)
+        {
+            SetLabels();
+            LoadDynamoView();
+            if (webView != null)
+            {
+                webView.NavigationCompleted -= WebView_NavigationCompleted;
+            }
+        }
+
+        private void LoadDynamoView()
+        {
+            DynamoModel model;
+            model = StartupUtils.MakeModel(false, ASMPath ?? string.Empty, hostAnalyticsInfo);
```

Comment:
The DynamoModel and DynamoViewModel currently are both hard coded to the one starting with default config here. This is fine for Sandbox but will be different in other cases, I would like to brainstorm with you to figure out how to pass in DynamoModel and DynamoViewModel and make things happen async. @RobertGlobant20 @filipeotero let's meet later